Sources/Target/Darwin/X86_64/ThreadX86_64.cpp

```diff
@@ -29,6 +29,16 @@ ErrorCode Thread::step(int signal, Address const &address) {
   }
   return resume(signal, address);
 }
+
+ErrorCode Thread::afterResume() {
+  ErrorCode error = modifyRegisters([](Architecture::CPUState &state) {
+    state.state64.gp.eflags &= ~(1 << 8);
+  });
+  if (error != kSuccess) {
+    return error;
+  }
+  return kSuccess;
```

Comment:
Indeed, the new version is better :)

On Wed, May 11, 2016 at 3:05 PM, kokarez <notifications@github.com> wrote:

> In Sources/Target/Darwin/X86_64/ThreadX86_64.cpp
> <https://github.com/facebook/ds2/pull/339#discussion_r62935238>:
>
> > @@ -29,6 +29,16 @@ ErrorCode Thread::step(int signal, Address const &address) {
> >    }
> >    return resume(signal, address);
> >  }
> > +
> > +ErrorCode Thread::afterResume() {
> > +  ErrorCode error = modifyRegisters([](Architecture::CPUState &state) {
> > +    state.state64.gp.eflags &= ~(1 << 8);
> > +  });
> > +  if (error != kSuccess) {
> > +    return error;
> > +  }
> > +  return kSuccess;
>
> Wait this check is useless!
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/339/files/c03f57caaf4695f786449526fd3da0b687ba2487#r62935238>
>



-- 
Stephane Sezer